core/src/main/java/greencity/service/impl/CloudStorageServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package greencity.service.impl;
+
+import com.google.cloud.WriteChannel;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import greencity.service.FileService;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+@Service
+public class CloudStorageServiceImpl implements FileService {
+    final String projectId;
+    final StorageOptions.Builder optionsBuilder;
+    private final String bucketName;
+    private final Storage storage;
+
+    /**
+     * Constructor with parameters.
+     */
+    public CloudStorageServiceImpl(@Value("${PROJECT_ID}") final String projectId,
+                                   @Value("${BUCKET_NAME}") final String bucketName) {
+        this.projectId = projectId;
+        this.bucketName = bucketName;
+        this.optionsBuilder = StorageOptions.newBuilder();
+        if (bucketName == null || bucketName.isEmpty()) {
+            throw new IllegalArgumentException("Bucket name is null");
+        }
+        if (projectId != null && !projectId.isEmpty()) {
+            optionsBuilder.setProjectId(projectId);
+        }
```

Comment:
BucketName and ProjectID can't be null because you are taking them from property file

Reply:
Also, please use Lombok annotation @RequiredArgsConstructor for creating constructors. Write constructor by yourself only if you need specific one